library/elasticbeanstalk_env.py

```diff
@@ -339,8 +342,7 @@ def main():
                                        option_settings=option_setting_tups,
                                        options_to_remove=None)
 
-                wait_for(ebs, app_name, env_name, wait_timeout, health_is_grey)
-                env = wait_for(ebs, app_name, env_name, wait_timeout, status_is_ready)
+                env = wait_for(ebs, app_name, env_name, wait_timeout, lambda environment: status_is_ready(environment) and version_is_updated(version_label, environment))
```

Comment:
What about the case where beanstalk downgrades to the previous version due to a deployment issue?